java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TestClassGenerator.java

```diff
@@ -0,0 +1,324 @@
+                for (final Lookup.Item<TestCreatorProvider> provider : providers) {
+                    if (provider.getDisplayName().equals(testingFramework)) {
+                        final TestCreatorProvider.Context context = new TestCreatorProvider.Context(new FileObject[]{fileObject});
+                        context.setSingleClass(true);
+                        context.setTargetFolder(targetFolder);
+                        context.setTestClassName(getPreffiledName(fileObject, testingFramework));
+                        FileChangeListener fcl = new FileChangeAdapter() {
+                            @Override
+                            public void fileDataCreated(FileEvent fe) {
+                                RequestProcessor.getDefault().post(() -> {
+                                    client.showDocument(new ShowDocumentParams(Utils.toUri(fe.getFile())));
+                                }, 1000);
+                            }
+                        };
+                        targetFolder.addRecursiveListener(fcl);
+                        try {
+                            provider.getInstance().createTests(context);
+                        } finally {
+                            RequestProcessor.getDefault().post(() -> {
+                                targetFolder.removeRecursiveListener(fcl);
+                            }, 1000);
+                        }
+                    }
+                }
+            } else {
+                throw new IllegalArgumentException(String.format("Illegal number of arguments received for command: %s", command));
+            }
+        } catch (JsonSyntaxException | IllegalArgumentException | MalformedURLException ex) {
+            client.showMessage(new MessageParams(MessageType.Error, ex.getLocalizedMessage()));
+        }
+        return CompletableFuture.completedFuture(true);
+    }
+
+    private static String getLocationText(Object location) {
+	String text = location instanceof SourceGroup
+		? ((SourceGroup) location).getDisplayName()
+		: location instanceof FileObject
+		? FileUtil.getFileDisplayName((FileObject) location)
+		: location.toString();
+	return text;
+    }
+
+    private static Map<Object, List<String>> getValidCombinations(CompilationInfo info, String methodName) {
+	List<String> testingFrameworks = getTestingFrameworks(info.getFileObject());
+	if (testingFrameworks.isEmpty()) {
+	    return null;
+	}
+	Map<Object, List<String>> validCombinations = new HashMap<>();
+	for (Object location : getLocations(info.getFileObject())) {
+	    String targetFolderPath = getTargetFolderPath(location);
+	    List<String> framework2Add = new ArrayList<>();
+	    for (String framework : testingFrameworks) {
+		String preffiledName = getPreffiledName(info.getFileObject(), framework);
+		preffiledName = preffiledName.replace(".", "/").concat(".java");
+		String path = targetFolderPath.concat("/").concat(preffiledName);
```

Comment:
Maybe use `File.separatorChar` here, if passing the path to JDK `File`.

Reply:
Fixed.